### PR TITLE
Hide 'updateState' method of AbstractVersionableEntity

### PR DESCRIPTION
**Pull request by vlad-lubenskyi**
Addresses https://github.com/SpineEventEngine/core-java/issues/782

This PR makes `updateState()` package-private and deals with consequences:

- `EntityBuilder` becomes `abstract`;
- `Given.entityOfClass(...)` method is removed;
- `GivenTest.createEntityBuilder()` test case is removed;
- `RecordBasedRepositoryTest` delegates entity creation to the subclasses (which use `EntityBuilder` implementation);
- `ColumnTest.invokeGetter` now works with `Aggregate` instead of an abstract entity;
- `EntityTest.ProvidesSuchEqualsThat.equalsToSame` now works with `Aggregate` instead of an abstract entity.

Spine version is advanced to 0.10.60-SNAPSHOT

### CODE REVIEW

**Comment by vlad-lubenskyi** [comment]
@armiol PTAL

File: server/src/test/java/io/spine/server/entity/RecordBasedRepositoryTest.java [hunk 1]
```diff
@@ -301,6 +295,8 @@ void noEntitiesIfEmpty() {
         }
     }
 
+    protected abstract E entity(I id);
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/RecordBasedRepositoryTest.java]: Please move to the `protected` methods section (see above). Also, I don't understand why it is not an overload of `createEntity() : E` method, which already exists above. `createEntity(id) : E` would look much more consistent.

File: server/src/test/java/io/spine/server/entity/given/EntityTestEnv.java [hunk 2]
```diff
@@ -100,4 +104,35 @@ public EntityWithMessageId() {
...
     }
+
+
+    public static class TestAggregate extends Aggregate<String, StringValue, StringValueVBuilder> {
+
+        protected TestAggregate(String id) {
+            super(id);
+        }
+
+        public static TestAggregate withStateOf(TestAggregate entity) {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/given/EntityTestEnv.java]: It's not only a "state", it's a copy. Therefore I'd call it `copyOf(...)`.

File: server/src/test/java/io/spine/server/entity/storage/given/EntityRecordWithColumnsTestEnv.java [hunk 3]
```diff
@@ -47,4 +49,12 @@ public EntityWithoutColumns(String id) {
             super(id);
         }
     }
+
+    public static class TestEntityBuilder extends EntityBuilder<TestEntity, String, Project> {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/storage/given/EntityRecordWithColumnsTestEnv.java]: There are two similar classes in this changeset. Another one is in `TestEntity`. Not sure we need both.

File: server/src/test/java/io/spine/server/entity/TestEntity.java [hunk 4]
```diff
@@ -34,32 +35,40 @@
...
     static TestEntity withStateOf(TestEntity entity) {
-        TestEntity result = Given.entityOfClass(TestEntity.class)
-                                 .withId(entity.getId())
-                                 .withState(entity.getState())
-                                 .withVersion(entity.getVersion()
+        TestEntity result = new TestEntityBuilder().setResultClass(TestEntity.class)
+                                                   .withId(entity.getId())
+                                                   .withState(entity.getState())
+                                                   .withVersion(entity.getVersion()
                                                     .getNumber())
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/TestEntity.java]: I think the formatting is broken for this line.

File: server/src/test/java/io/spine/server/entity/given/EntityTestEnv.java [hunk 5]
```diff
@@ -100,4 +104,35 @@ public EntityWithMessageId() {
             super(Sample.messageOfType(ProjectId.class));
         }
     }
+
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/given/EntityTestEnv.java]: This line is redundant.

**Comment by armiol** [comment]
@vlad-lubenskyi also before merging, please be careful and make sure that the bumped version in this PR is indeed an increment in relation to `master`. There is a concurrent PR which also has `60`.

**Comment by vlad-lubenskyi** [comment]
@armiol PTAL

File: server/src/test/java/io/spine/server/entity/given/EntityTestEnv.java [hunk 6]
```diff
@@ -100,4 +104,34 @@ public EntityWithMessageId() {
...
+    public static class TestAggregate extends Aggregate<String, StringValue, StringValueVBuilder> {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/given/EntityTestEnv.java]: Please create an issue saying that currently we use aggregates in tests for `Entity`, but it seems odd. There should be a cleaner way of doing it. And put the reference to the issue as a todo item here.

**Comment by vlad-lubenskyi** [comment]
@armiol PTAL

**Pull request merged**